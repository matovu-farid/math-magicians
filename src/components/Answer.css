.answer {
  text-align: end;
  width: 280px;
  padding: 15px ;
  text-indent: -35px;
  background-color: #858694;
  color: white;
}