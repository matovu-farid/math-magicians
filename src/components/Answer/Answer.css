.answer {
  text-align: end;
  width: 280px;
  padding: 10px;
  text-indent: -35px;
  background-color: #858694;
  color: white;
}
