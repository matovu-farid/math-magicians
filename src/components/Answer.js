import './Answer.css';

const Answer = () => <div className="answer">0</div>;

export default Answer;
