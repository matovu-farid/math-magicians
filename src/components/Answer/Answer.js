import propTypes from 'prop-types';
import { Component } from 'react';
import './Answer.css';

class Answer extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { total, next, operation } = this.props;
    if (total === null && next === null) return (<div className="answer">0</div>);
    if (total === null) return (<div className="answer">{next}</div>);
    return (<div className="answer">{`${total} ${operation ?? ''} ${next ?? ''}`}</div>);
  }
}

Answer.propTypes = {
  total: propTypes.string,
  next: propTypes.string,
  operation: propTypes.string,
};

Answer.defaultProps = {
  total: '',
  next: '',
  operation: '',
};

export default Answer;
