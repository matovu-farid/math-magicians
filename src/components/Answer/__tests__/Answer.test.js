import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Answer from '../Answer';

describe.skip('Answer component tests', () => {
  let sampleObject;
  let nullObject;

  beforeEach(() => {
    sampleObject = {
      total: '10',
      next: '2',
      operation: '+',
    };
    nullObject = {
      total: null,
      next: null,
      operation: null,
    };
  });

  it('should display "10 + 2"', () => {
    const { total, next, operation } = sampleObject;
    render(<Answer total={total} next={next} operation={operation} />);
    const displayText = screen.queryByText('10 + 2');
    expect(displayText).toBeInTheDocument();
  });
  it('should display "0" with no input', () => {
    const { total, next, operation } = nullObject;
    render(<Answer total={total} next={next} operation={operation} />);
    const displayText = screen.queryByText('0');
    expect(displayText).toBeInTheDocument();
  });
  it('should render the value typed by the user', () => {
    const { total, operation } = nullObject;
    render(<Answer total={total} next={7} operation={operation} />);
    const displayText = screen.queryByText('7');
    expect(displayText).toBeInTheDocument();
  });
});
