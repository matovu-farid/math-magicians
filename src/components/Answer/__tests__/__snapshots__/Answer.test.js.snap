// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer component tests renders correctly 1`] = `
<div
  className="answer"
>
  10 + 2
</div>
`;
